yb-voyager/cmd/exportSchema.go

```diff
@@ -331,6 +336,82 @@ func applyMigrationAssessmentRecommendations() error {
 	return nil
 }
 
+// TODO: merge this function with applying sharded/colocated recommendation
+func applyMergeConstraintsTransformations() {
+	if utils.GetEnvAsBool("YB_VOYAGER_SKIP_MERGE_CONSTRAINTS_TRANSFORMATIONS", false) {
+		log.Infof("skipping applying merge constraints transformation due to env var YB_VOYAGER_SKIP_MERGE_CONSTRAINTS_TRANSFORMATIONS=true")
+		return
+	}
+
+	utils.PrintAndLog("Applying merge constraints transformation to the exported schema")
+	transformer := transform.NewTransformer()
+
+	fileName := utils.GetObjectFilePath(schemaDir, TABLE)
+	if !utils.FileOrFolderExists(fileName) {
+		log.Infof("table.sql file doesn't exists, skipping applying merge constraints transformation")
+		return
+	}
+
+	rawStmts, err := queryparser.ParseSqlFile(fileName)
+	if err != nil {
+		utils.ErrExit("failed to parse table.sql file: %v", err)
+	}
+
+	// filter SET statements(PostgreSQL) from rawStmts
+	tableStmts := lo.Filter(rawStmts.Stmts, func(stmt *pg_query.RawStmt, _ int) bool {
+		stmtType := queryparser.GetStatementType(stmt.Stmt.ProtoReflect())
+		return stmtType != queryparser.PG_QUERY_VARIABLE_SET_STMT_NODE && stmtType != queryparser.PG_QUERY_SELECTSTMT_NODE
+	})
+
+	// TODO: separate handling for Oracle and MySQL ??
+	sessionVarStmts := lo.Filter(rawStmts.Stmts, func(stmt *pg_query.RawStmt, _ int) bool {
+		stmtType := queryparser.GetStatementType(stmt.Stmt.ProtoReflect())
+		return stmtType == queryparser.PG_QUERY_VARIABLE_SET_STMT_NODE || stmtType == queryparser.PG_QUERY_SELECTSTMT_NODE
+	})
+
+	// NOTE: In case of any error during merge constraints or deparsing, log the error and stick to the original table.sql file
+
+	transformedRawStmts, err := transformer.MergeConstraints(tableStmts)
+	if err != nil {
+		log.Errorf("failed to merge constraints: %v", err)
+		return
+	}
+
+	sessionVarsSqls, err := queryparser.DeparseRawStmts(sessionVarStmts)
```

Comment:
Idk if this is easy to do, but for all statements that are unchanged, can we retain the original statement/formatting.

Reply:
Here we are dealing with sql file level for even parsing.
In this code path we don't have formatted sql statements(which we get using our regexp based parsing logic).

Reply:
I wanted to fix that in a different manner - Introduce a formatter to format this 1 line ddl into formatted sql.
Solution is still not known.

Already have a ticket for that.

Reply:
Hmm That we can do, let me think about it.

Reply:
btw @priyanshi-yb we will be merging PK, Unique and Check constraint. 
Even if a single stmt of any of those type is there the file will be updated. so do you think it will be valuable?
nevertheless i have added your suggestion to the ticket(https://yugabyte.atlassian.net/browse/DB-15410) so that we dont miss it.